src/store/actions.js

```diff
@@ -38,29 +38,32 @@ export default {
     if (ids.length) {
       return fetchItems(ids)
         .then(items => {
-          if (items.every(item => item.type === 'story')) {
+          // Don't fetch similar if all items are Jobs (on /jobs)
+          // In other tabs there is rarely many jobs
+          let stories = items.filter(x => x.type === 'story')
+          let otherItems = items.filter(x => x.type !== 'story')
 
-            return fetchSimilar(items.map(item => item.title))
-              .then(similar => items.map((item, idx) => {
-                item.similar = similar[idx];
-                return item;
-              }))
-              // Start fetching similar posts (potential performance issue...)
-              .then((items) => {
-                return fetchItems(items.map(i => i.similar).flat().map(sim => sim.id))
-                  .then(similarItems => {
-                    items.forEach(item => {
-                      item.similar = item.similar.map(sim => {
-                        const simItem = similarItems.find(si => si.id === sim.id);
-                        return Object.assign({ similarity_score: sim.score }, simItem);
-                      });
+          return fetchSimilar(stories.map(item => item.title))
+            .then(similar => items.map((item, idx) => {
```

Comment:
you're potentially mapping the wrong similar posts to the wrong items here since you only fetch similar posts for stories and then map over all items och assign the similar posts to the first n items.